gadget/update.go

```diff
@@ -366,6 +390,11 @@ type Updater interface {
 }
 
 func applyUpdates(new GadgetData, updates []updatePair, rollbackDir string, observer ContentUpdateObserver) error {
+	// XXX: too simplistic? should we pass the
+	// ResolvedContentFilterFunc from updatePair down to the
+	// MountedFileSystem instead?
+	filterUpdates(updates)
```

Comment:
I think this might be ok for now, and we can revisit later, at least is not a blocker